gitops/snapshot.go

```diff
@@ -775,3 +775,33 @@ func ResetSnapshotStatusConditions(adapterClient client.Client, ctx context.Cont
 
 	return nil
 }
+
+// CopySnapshotLabelsAndAnnotation coppies labels and annotations from build pipelineRun or tested snapshot
+// into regular or composite snapshot
+func CopySnapshotLabelsAndAnnotation(application *applicationapiv1alpha1.Application, snapshot *applicationapiv1alpha1.Snapshot, componentName string, source *metav1.ObjectMeta, prefix string, isComposite bool) {
```

Comment:
I think we'll need [unit test coverage](https://app.codecov.io/gh/redhat-appstudio/integration-service/pull/513) for this new function.

Reply:
Added unit tests for covering newly implemented function.

Reply:
Now the [ginkgo linter is complaining](https://github.com/redhat-appstudio/integration-service/actions/runs/7528084698/job/20489547180?pr=513) - it's about the `Expect(err).To(BeNil())` calls that should be changed to `Expect(err).ToNot(HaveOccurred())`.